### PR TITLE
Morganey API

### LINKED ISSUE #199: Morganey API
To be able to use Morganey in the implementation of the online REPL a few refactorings should be done in order to achieve a cleaner Morganey API.
## 

Proposed refactorings:
- Decouple autocompletion from terminal-REPL (jLine). This could be done by transforming the [`ReplAutocompletion`](https://github.com/rexim/Morganey/blob/db15b3b438ce722167a882a4cecdbfab7162d5de/src/main/scala/me/rexim/morganey/ReplAutocompletion.scala#L17) class into an object and removing the inheritance from `jline.console.completer.Completer`.
- Decouple REPL-logic from terminal based user interface (jLine). (Handle parsing of a line, which was typed in in object [`MorganeyRepl`](https://github.com/rexim/Morganey/blob/db15b3b438ce722167a882a4cecdbfab7162d5de/src/main/scala/me/rexim/morganey/interpreter/MorganeyRepl.scala#L10) instead of [`Main`](https://github.com/rexim/Morganey/blob/db15b3b438ce722167a882a4cecdbfab7162d5de/src/main/scala/me/rexim/morganey/Main.scala#L73).)

### PR DESCRIPTION
**Pull request by keddelzz**
Prepare Morganey to be used via API (e.g. in the online REPL #182)
## 

closes #199

### CODE REVIEW

File: src/main/scala/me/rexim/morganey/interpreter/MorganeyRepl.scala [hunk 1]
```diff
@@ -1,13 +1,37 @@
...
+    line.trim match {
+      case ""            => Computation(ReplResult(context))
+      case Commands(cmd) => Computation(cmd(context))
+      case input         => parseAndEval(context, line)
+    }
+
+  private def parseAndEval(context: ReplContext, line: String): Computation[ReplResult[String]] = {
+    val parseResult = Computation(LambdaParser.parseWith(line, _.replCommand))
+
+    // TODO(#197): discriminate bindings from the rest of the nodes here and inform the user if the binding was redefined
```
Comment (rexim) [inline src/main/scala/me/rexim/morganey/interpreter/MorganeyRepl.scala:26]: Just for the sake of issue linking: TODO(#197) was moved here.

**Review by rexim — CHANGES_REQUESTED** [review]
Manual testing detected a REPL bug:

```
λ> load prelude
λ> pair
Non-existing binding: pair
λ> pair 1 0
Non-existing binding: pair
λ> 
```

On the current master everything works as expected:

```
λ> load prelude
λ> pair
term: λx.y.z.z x y
λ> pair 1 0
numbers: [1]
λ> 
```

**Comment by keddelzz** [comment]
Wow, what a stupid bug :laughing:

**Review by rexim — APPROVED** [review]

**Comment by rexim** [comment]
@keddelzz Wow! That's an interesting situation indeed! It was also pretty hard to avoid. It feels like it's a nature of impertive impure approach we are using here. Very interesting...

**Comment by rexim** [comment]
@ForNeVeR sorry we wanna merge it right now, I guess. Please feel free to leave your comments even after the merge. We will convert them into issues.

**Comment by rexim** [comment]
@keddelzz thanks for the changes!

**Comment by ForNeVeR** [comment]
Sorry guys I wasn't available and not still fully conscious at the moment. Will review everything later.

**Comment by rexim** [comment]
@ForNeVeR no worries! Take your time!

**Review by ForNeVeR — COMMENTED** [review]
As for the changes they LGTM 👍

**Comment by rexim** [comment]
@ForNeVeR cool! Thank you very much!

**Pull request merged**